src/library_wasmfs.js

```diff
@@ -392,7 +395,60 @@ FS.createPreloadedFile = FS_createPreloadedFile;
         return __wasmfs_mknod(pathBuffer, mode, dev);
       }));
     },
-    // TODO: mkdev
+    major: (dev) => ((dev) >> 8),
+    minor: (dev) => ((dev) & 0xff),
+    makedev: (ma, mi) => ((ma) << 8 | (mi)),
+    registerDevice(dev, ops) {
+      var backendPointer = _wasmfs_create_jsimpl_backend();
+
+      var defaultOps = {
+        allocFile: (file) => {},
+        freeFile: (file) => {},
+        write: (file, buffer, length, offset) => {},
+        read: (file, buffer, length, offset) => {},
+        getSize: (file) => {}
+      }
+
+      if (typeof ops.allocFile === 'undefined') {
+        ops.allocFile = defaultOps.allocFile;
+      }
+
+      if (typeof ops.freeFile === 'undefined') {
+        ops.freeFile = defaultOps.freeFile;
+      }
+
+      if (typeof ops.write === 'undefined') {
+        ops.write = defaultOps.write;
+      }
+
+      if (typeof ops.read === 'undefined') {
+        ops.read = defaultOps.read;
+      }
+
+      if (typeof ops.getSize === 'undefined') {
+        ops.getSize = defaultOps.getSize;
+      }
```

Comment:
A more compact way to do this is to use [`Object.assign`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign).